components/src/dynamo/vllm/omni/stage_router.py

```
@@ -38,13 +49,26 @@ def __init__(
+        resolved_stage_configs_path, self.stage_configs = load_and_resolve_stage_configs(
             config.model,
             stage_configs_path,
             kwargs={},
         )
         self.stage_clients: Dict[str, Any] = {}
 
+        # Initialize connectors so the router can fetch final-stage output
+        # via connector.get() instead of SHM — enabling multi-node deployments.
+        try:
+            register_dynamoomni_nixl_connector()
+        except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)